src/presentation/http/router/noteSettings.ts

```diff
@@ -173,6 +174,43 @@ const NoteSettingsRouter: FastifyPluginCallback<NoteSettingsRouterOptions> = (fa
     return reply.send(team);
   });
 
+  /**
+   * Generates a new invitation hash for a specific note
+   */
+  fastify.patch<{
+    Params: {
+      notePublicId: NotePublicId;
+    },
+    Reply: InvitationHash,
+  }>('/:notePublicId/invitation-hash', {
+    config: {
+      policy: [
+        'authRequired',
+        'userInTeam',
+      ],
+    },
+    schema: {
+      params: {
+        notePublicId: {
+          $ref: 'NoteSchema#/properties/id',
+        },
+      },
+    },
+    preHandler: [
+      noteResolver,
+    ],
+  }, async (request, reply) => {
+    const noteId = request.note?.id as number;
+
+    const updatedNoteSettings = await noteSettingsService.regenerateInvitationHash(noteId);
+
+    if (updatedNoteSettings === null) {
+      return reply.notFound('Note settings not found');
```

Comment:
I think 404 code is not correct here, since we have an exception and we need to return corresponded code.

Reply:
it is still actual

Reply:
Check please